lib/catalog/owid/catalog/tables.py

```diff
@@ -540,3 +588,638 @@ def _repr_html_(self):
+        )
+
+    if not inplace:
+        return to_table
+
+
+@overload
+def update_log(
+    table: Table,
+    operation: str,
+    parents: Optional[List[Any]] = None,
+    variable_names: Optional[List[str]] = None,
+    comment: Optional[str] = None,
+    inplace: bool = True,
+) -> None:
+    ...
+
+
+@overload
+def update_log(
+    table: Table,
+    operation: str,
+    parents: Optional[List[Any]] = None,
+    variable_names: Optional[List[str]] = None,
+    comment: Optional[str] = None,
+    inplace: bool = False,
+) -> Table:
+    ...
+
+
+def update_log(
+    table: Table,
+    operation: str,
+    parents: Optional[List[Any]] = None,
+    variable_names: Optional[List[str]] = None,
+    comment: Optional[str] = None,
+    inplace: bool = False,
+) -> Optional[Table]:
+    if not inplace:
+        table = table.copy()
+
+    # Append a new entry to the processing log of the required variables.
+    if variable_names is None:
+        # If no variable is specified, assume all (including index columns).
+        variable_names = list(table.all_columns)
+    for column in variable_names:
+        # If parents is not defined, assume the parents are simply the current variable.
+        _parents = parents or [column]
+        # Update (in place) the processing log of current variable.
+        table._fields[column].processing_log = variables.add_entry_to_processing_log(
+            processing_log=table._fields[column].processing_log,
+            variable_name=column,
+            parents=_parents,
+            operation=operation,
+            comment=comment,
+        )
+
+    if not inplace:
+        return table
+
+
+def amend_log(
+    table: Table,
+    operation: str,
+    parents: Optional[List[Any]] = None,
+    variable_names: Optional[List[str]] = None,
+    comment: Optional[str] = None,
+    entry_num: Optional[int] = -1,
+    inplace: bool = False,
+) -> Optional[Table]:
+    if not inplace:
+        table = table.copy()
+
+    # Append a new entry to the processing log of the required variables.
+    if variable_names is None:
+        # If no variable is specified, assume all (including index columns).
+        variable_names = list(table.all_columns)
+    for column in variable_names:
+        # If parents is not defined, assume the parents are simply the current variable.
+        _parents = parents or [column]
+        # Update (in place) the processing log of current variable.
+        table._fields[column].processing_log = variables.amend_entry_in_processing_log(
+            processing_log=table._fields[column].processing_log,
+            variable_name=column,
+            parents=_parents,
+            operation=operation,
+            comment=comment,
+            entry_num=entry_num,
+        )
+
+    if not inplace:
+        return table
+
+
+def get_unique_sources_from_table(table: Table) -> List[Source]:
```

Comment:
Dataclasses support equality operator, so instead of converting to dict and then to objects again (which could affect performance), you could do something like this

```
unique_sources = []
for source in sources:
    if source not in unique_sources:
        unique_sources.append(source)
```

(not urgent, can be done in a different PR)

Reply:
New issue: https://github.com/owid/etl/issues/1270